core/src/components/button/test/form-reference/button.e2e.ts

```diff
@@ -0,0 +1,93 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('button: form', () => {
+  test('should submit the form by id', async ({ page }) => {
+    await page.setContent(`
+      <form id="myForm"></form>
+      <ion-button form="myForm" type="submit">Submit</ion-button>
+    `);
+
+    const submitEvent = await page.spyOnEvent('submit');
+
+    await page.click('ion-button');
+
+    expect(submitEvent).toHaveReceivedEvent();
+  });
+
+  test('should submit the form by reference', async ({ page }) => {
+    await page.setContent(`
+      <form></form>
+      <ion-button type="submit">Submit</ion-button>
+      <script>
+        const form = document.querySelector('form');
+        const button = document.querySelector('ion-button');
+        button.form = form;
+      </script>
+    `);
+
+    const submitEvent = await page.spyOnEvent('submit');
+
+    await page.click('ion-button');
+
+    expect(submitEvent).toHaveReceivedEvent();
+  });
+
+  test('should submit the closest form', async ({ page }) => {
+    await page.setContent(`
+      <form>
+         <ion-button type="submit">Submit</ion-button>
+      </form>
+    `);
+
+    const submitEvent = await page.spyOnEvent('submit');
+
+    await page.click('ion-button');
+
+    expect(submitEvent).toHaveReceivedEvent();
+  });
+
+  test.describe('should throw a warning if the form cannot be found', () => {
+    test('form is a string selector', async ({ page }) => {
+      await page.setContent(`<ion-button type="submit" form="missingForm">Submit</ion-button>`);
+
+      const logs: string[] = [];
+
+      page.on('console', (msg) => {
+        logs.push(msg.text());
+      });
+
+      await page.click('ion-button');
+
+      expect(logs.length).toBe(1);
+      expect(logs[0]).toContain(
+        '[Ionic Warning]: Form with selector: "#missingForm" could not be found. Verify that the selector is correct and the form is rendered in the DOM.'
```

Comment:
This check seems brittle. The moment we change the contents of the warning, this test will fail. Are you trying to check that a warning is logged or are you trying to check what the contents of the warning are?

Reply:
Both. Although I could just check for `[Ionic Warning]` if we are concerned with the message contents being updated and needing to update the test over time.

Reply:
I'm fine keeping this as is for now. Long term it may be good to have a warning/error ID like "MISSING_FORM_ELEMENT" that we could check, but that needs design work first and should not block this PR.